gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-healthcheck/src/main/java/io/gravitee/gateway/services/healthcheck/rule/EndpointRuleHandler.java

```diff
@@ -117,13 +118,15 @@ public EndpointRuleHandler(Vertx vertx, EndpointRule<T> rule, TemplateEngine tem
 
     @Override
     public void handle(Long timer) {
+        MDC.put("api", rule.api().getId());
         T endpoint = rule.endpoint();
         logger.debug("Running health-check for endpoint: {} [{}]", endpoint.getName(), endpoint.getTarget());
 
         // Run request for each step
         for (HealthCheckStep step : rule.steps()) {
             runStep(endpoint, step);
         }
+        MDC.remove("api");
```

Comment:
Encapsulate with a `try {} finally{}` so we are sure the API ID is not in the MDC context at the end.